specs/agents/metadata.md

```diff
@@ -75,6 +76,13 @@ hostname if `configured_hostname` is not provided.
 Agents that are APM-Server-version-aware, or that are compatible only with versions >= 7.4, should
 use the new fields wherever applicable.
 
+#### Host.id
+
+Agents MAY collect the `host.id` as an unique identifier for the host.
```

Comment:
As we have two agents involved here (the APM agent and the profiling host agent) we should always qualify which agent we actually mean. Also there are inconsistencies in terminology throughout the document. We call the APM agent:

* APM agent
* APM-agent
* agent

and the profiling host agent:

* profiling host agent
* profiler host agent
* host agent

Reply:
I've clarified the "agent" usages in this section.
For the rest of the document, the inconsistencies come from the fact that this document was originally intended as only an APM-agent spec. This metadata spec is also becoming less important as we move to Otel SemConv, which effectively will replace this spec.

I'm just adding the `host.id` here, because we intend to also add the universal profiling integration to the old `elastic-apm-agent java` to allow easier adoption for existing users.